requirements.txt

```diff
@@ -1,42 +1,121 @@
-asgiref==3.2.10
-boto3==1.17.90
-certifi==2020.12.5
-channels==3.0.3
-channels_redis==3.2
-chardet==3.0.4
-device-detector==0.10
-Django==3.1.1
-django-cors-headers==3.7.0
-django-request-logging==0.7.2
-django-s3-storage==0.13.4
-django-storages==1.11.1
-djangorestframework==3.12.2
-idna==2.10
-ipdb==0.13.4
-jmespath==0.10.0
-numpy==1.20.1
-pandas==1.2.2
-psycopg2-binary==2.8.6
-python-dateutil==2.8.1
-pytz==2020.1
-PyYAML==5.4.1
-regex==2020.11.13
-requests==2.24.0
-six==1.15.0
-sqlparse==0.4.1
-tqdm==4.51.0
-urllib3==1.25.11
-Pillow==8.1.2
-django-tenants==3.2.1
-djangorestframework==3.12.2
-Markdown==3.3.4
-django-filter==2.4.0
-drf-yasg==1.20.0
-django-safedelete==1.0.0
-django-oauth2-provider==0.2.6.1
-django-rest-framework-social-oauth2==1.1.0
-coverage==5.5
-google-cloud-bigquery==2.20.0
-pyarrow==4.0.1
-sentry-sdk==1.1.0
-django-silk==4.1.0
+aioredis==1.3.1
+asgiref==3.2.10
+async-timeout==3.0.1
+attrs==21.2.0
+autobahn==21.3.1
+Automat==20.2.0
+autopep8==1.5.7
+backcall==0.2.0
+boto3==1.17.90
+botocore==1.20.112
+cachetools==4.2.2
+certifi==2020.12.5
+cffi==1.14.6
+channels==3.0.3
+channels-redis==3.2.0
+chardet==3.0.4
+constantly==15.1.0
+coreapi==2.3.3
+coreschema==0.0.4
+coverage==5.5
+cryptography==3.4.8
+daphne==3.0.2
+decorator==5.1.0
+defusedxml==0.7.1
+Deprecated==1.2.13
+device-detector==0.10
+Django==3.1.1
+django-braces==1.14.0
+django-cors-headers==3.7.0
+django-filter==2.4.0
+django-oauth-toolkit==1.5.0
+django-oauth2-provider==0.2.6.1
+django-redis==5.0.0
+django-request-logging==0.7.2
+django-rest-framework-social-oauth2==1.1.0
+django-s3-storage==0.13.4
+django-safedelete==1.0.0
+django-silk==4.1.0
+django-storages==1.11.1
+django-tenants==3.2.1
+django-utils-six==2.0
+djangorestframework==3.12.2
+drf-cached-instances==0.3.4
```

Comment:
Lots of new dependencies? :O

Reply:
Not really. I ran `pip freeze > requirements.txt` and it loaded the sub-dependencies also to this file and then sorted them.

It's more or less same. Let me revert. I think it'll be difficult if we need to clear sub-dependencies when we're removing any dependency.

Reply:
@rathorevaibhav use this: https://github.com/bndr/pipreqs

this will not list the sub-dependencies.